### PR TITLE
seek by decimal keys

### PR DESCRIPTION
**Pull request by yudukikun5120**
You will be able to seek videos at MS Streams or TED by pressing the numerical keys: `0-9` with this feature.
I tried to work on Amazon Video, but it did not achieve.

### CODE REVIEW

**Comment by yudukikun5120** [comment]
@mkobayashime I do not subscribe to DAZN, so could you add snippets below to `dazn.js` and check it if you could?

```js
case "0":
case "1":
case "2":
case "3":
case "4":
case "5":
case "6":
case "7":
case "8":
case "9": {
  if (this.config[`keys-decimal`]) {
    decimalSeek({
      media: this.media,
      numericKey: e.key,
    })
    this.callIndicatorCreator({
      type: "text",
      text: e.key,
    })
  }
  break
}
```

**Comment by mkobayashime** [comment]
@yudukikun5120 
It's not working in DAZN, and I couldn't find any solution for that.
As for Prime Video I changed a few lines to make this work #100.

File: public/options.html [hunk 1]
```diff
@@ -260,6 +260,27 @@ <h2>Enabled Shortcuts</h2>
               <div class="checkbox-style" />
             </div>
           </li>
+          <li>
+            <div class="keys-outer">
+              <div class="keys-spacer">
+                <span class="key">0-9</span>
```
Comment (mkobayashime) [inline public/options.html:266]:
```suggestion
                <span class="key">0</span>
                –
                <span class="key">9</span>
```

File: src/methods/seek.js [hunk 2]
```diff
@@ -29,6 +29,17 @@ const seek = ({ media, direction, seekSec, cacheRequired = false }) => {
   }
 }
 
+/**
+ * @param {HTMLVideoElement} media - Video element to be handled
+ * @param {number} numericKey - Number key pressed. Must be between 0 and 9
```
Comment (mkobayashime) [inline src/methods/seek.js:34]:
```suggestion
 * @param {string} numericKey - Number key pressed. Must be between 0 and 9
```

Values passed are string actually.

File: src/methods/seek.js [hunk 3]
```diff
@@ -29,6 +29,17 @@ const seek = ({ media, direction, seekSec, cacheRequired = false }) => {
...
+ * @param {boolean} [cacheRequired=true] - `true` to force caching the new frame
+ * @returns {void}
+ */
+const decimalSeek = ({ media, numericKey, cacheRequired = true }) => {
+  media.currentTime = media.duration * (numericKey / 10)
```
Comment (mkobayashime) [inline src/methods/seek.js:39]:
```suggestion
  const keyIndex = parseInt(numericKey)
  if (isNaN(keyIndex)) return

  media.currentTime = media.duration * (keyIndex / 10)
```

We should avoid implicit casting.

**Review by mkobayashime — CHANGES_REQUESTED** [review]

File: src/methods/seek.js (hunk 2 above)
Comment (yudukikun5120) [reply src/methods/seek.js]: How shameful!

**Comment by yudukikun5120** [comment]
@mkobayashime I sincerely appreciate the rich change suggestions.

**Comment by yudukikun5120** [comment]
@mkobayashime I implemented the codes for Prime Video and asked you for a code review.

**Comment by mkobayashime** [comment]
@yudukikun5120 
Could you squash misc commits (especially ones simply applying suggestions by me, originally changed in f6d366340dae34d98710498d672d0c8285fb63ee) before I merge them?

**Comment by yudukikun5120** [comment]
@mkobayashime It would presumably be better to use _squash and merge_ because it marks the index of PR (e.g., `fix the hoge (#PRIndex)`) on the squashed commit message. Could you try it?

**Comment by mkobayashime** [comment]
@yudukikun5120 
I suggeted the squashing because I thought you were intending to merge these changes with multiple (not all-in-one) commits per intension, but I can do the _Squash and merge_ if it's OK for you.

**Comment by yudukikun5120** [comment]
@mkobayashime I appreciate your thoughtfulness. I'm OK with making it in a single commit.

**Pull request merged**